CMR/python/cmr/auth/token.py

```diff
@@ -0,0 +1,317 @@
+    return "sit."
+
+def uat():
+    """ User to return the URL part that points to UAT """
+    return "uat."
+
+def prod():
+    """ User to return the URL part that points to production """
+    return ""
+
+def _request_token(user, options):
+    """
+    Internal worker function to call CMR and request a token
+    Parameters:
+        user(string): EDL user name
+        password(string): EDL password
+        use_cache(boolean): True if tokens are to be cached and reused
+        client_name: HTTP client name to use in communications
+        env(string): CMR environment, defaults to prod
+        client_address(string): ip address of the machine making the request for token
+    Returns:
+        token or text of exceptions from HTTPError
+    """
+    #unpack
+    clear_password = options["clear_text_password"]
+    use_cache = options["cached"]
+    client_name = options["client_name"]
+    env = options["env"]
+    client_address = options["address"]
+
+    if use_cache:
+        token_text = _read_token_file()
+    else:
+        token_text = None
+    if token_text is not None and len(token)>0:
+        return token_text
+    env = env if env.endswith(".") else env + "."
+    data='<token>\
+<username>{}</username>\
+<password>{}</password>\
+<client_id>{}</client_id>\
+<user_ip_address>{}</user_ip_address>\
+</token>'.format(user, clear_password, client_name, client_address)
+
+    url='https://cmr.{}earthdata.nasa.gov/legacy-services/rest/tokens.json'.format(env)
+    encoded_data = str.encode(data)
+    req = urllib.request.Request(url, data=encoded_data)
+    req.add_header('Content-Type', 'application/xml')
+    req.add_header('User-Agent', client_name)
+    try:
+        resp = urllib.request.urlopen(req)
+        obj_json = json.loads(resp.read().decode('utf-8'))
+
+        token_text = obj_json['token']['id']
+        if use_cache:
+            _write_token_file(token_text)
+        return token_text
+    except urllib.error.HTTPError as exception:
+        return exception
+
+def token(edl_user_name, password_lambda=password_file, opts=None):
+    """
+    Requests an Earth Data Login Token from CMR
+    Parameters:
+        edl_user_name(string): EDL user name
+        password_lambda(lambda): function for returning password
+        opt(dictionary): dictionary of options
+    Returns:
+        token or text of exceptions from HTTPError
+    """
+
+    # handle the default, don't pass in bad data to the internal handler
+    if opts is None:
+        opts = {}
+    clean={"env": c.dict_or_default(opts, "cmr.env", prod()),
+        "address": c.dict_or_default(opts, 'client.address', _get_local_ip),
+        "client_name": c.dict_or_default(opts, 'client.name', 'python_cmr_lib'),
+        "cached": c.dict_or_default(opts, "cache.token", True),
+        "clear_text_password": password_lambda(id, opts)}
+
+    edl_token = _request_token(edl_user_name, clean)
+    #    clear_text_password,
```

Comment:
commented out code

Reply:
this has been removed, I don't know why it is showing up here